samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -642,4 +646,73 @@ public void testShouldStopStreamApplicationWhenShutdownTimeOutIsLessThanContaine
     assertEquals(ApplicationStatus.SuccessfulFinish, appRunner3.status());
   }
 
+  /**
+   * A. Create a kafka topic with partition count set to 5.
+   * B. Create and launch a samza application which consumes events from the kafka topic.
+   * C. Validate that the {@link JobModel} contains 5 {@link SystemStreamPartition}'s.
+   * D. Increase the partition count of the input kafka topic to 100.
+   * E. Validate that the new {@link JobModel} contains 100 {@link SystemStreamPartition}'s.
+   */
+  @Test
+  public void testShouldGenerateJobModelOnPartitionCountChange() throws Exception {
+    publishKafkaEvents(inputKafkaTopic, 0, NUM_KAFKA_EVENTS, PROCESSOR_IDS[0]);
+
+    // Create StreamApplication from configuration.
+    CountDownLatch kafkaEventsConsumedLatch1 = new CountDownLatch(NUM_KAFKA_EVENTS);
+    CountDownLatch processedMessagesLatch1 = new CountDownLatch(1);
+
+    ApplicationRunner appRunner1 = ApplicationRunners.getApplicationRunner(TestStreamApplication.getInstance(
+            TEST_SYSTEM, inputKafkaTopic, outputKafkaTopic, processedMessagesLatch1, null, kafkaEventsConsumedLatch1,
+            applicationConfig1), applicationConfig1);
+
+    appRunner1.run();
+
+    processedMessagesLatch1.await();
+
+    // Processor-1 has finished consuming all the messages.
+    kafkaEventsConsumedLatch1.await();
+
+    String jobModelVersion = zkUtils.getJobModelVersion();
+    JobModel jobModel = zkUtils.getJobModel(jobModelVersion);
+    Set<SystemStreamPartition> ssps = getSystemStreamPartitions(jobModel);
+
+    // Validate that the input partition count is 5 in the JobModel.
+    Assert.assertEquals(5, ssps.size());
+
+    // Increase the partition count of input kafka topic to 100.
+    AdminUtils.addPartitions(zkUtils(), inputKafkaTopic, 100, "", true, RackAwareMode.Enforced$.MODULE$);
+
+    publishKafkaEvents(inputKafkaTopic, 0, NUM_KAFKA_EVENTS, PROCESSOR_IDS[0]);
+
+    CountDownLatch kafkaEventsConsumedLatch2 = new CountDownLatch(NUM_KAFKA_EVENTS);
+    CountDownLatch processedMessagesLatch2 = new CountDownLatch(1);
+
+    ApplicationRunner appRunner2 = ApplicationRunners.getApplicationRunner(TestStreamApplication.getInstance(
```

Comment:
why do you need to start another processor? The monitor should trigger job model regeneration, shouldn't it?

Reply:
Will change it. Thanks for pointing it out.